nomad/resource_job.go

```diff
@@ -629,6 +648,29 @@ func resourceJobRead(d *schema.ResourceData, meta interface{}) error {
 		d.Set("allocation_ids", nil)
 	}
 
+	// Update jobspec submission data if available.
+	// Safely ignore errors as this is an optional step.
+	sub, _, err := client.Jobs().Submission(*job.ID, int(*job.Version), opts)
+	if err != nil {
+		log.Printf("[WARN] failed to read job submission: %v", err)
+	} else if sub != nil {
+		if sub.Source != "" {
+			d.Set("jobspec", sub.Source)
```

Comment:
> so remote HCL changes can be detected that the current `jobspecEqual()` might otherwise miss.

Yes, currently `jobspecEqual()` doesn't even detect any remote changes because, up until now, it didn't have access to the remote jobspec.

>  if semantically unimportant things (like hcl comments) change, then there still would be no change detected

Right, but that's not because of this change, this is because `jobspecEqual()` does a semantic comparison, meaning, it parses the raw string and compares the result job object, so changes like comments and formatting are ignored.
https://github.com/hashicorp/terraform-provider-nomad/blob/5afd8e70e7b927a9a1bb447f9ce4d0d543d8f307/nomad/resource_job.go#L989-L1032